OpenSim/Analyses/JointReaction.cpp

```diff
@@ -310,8 +310,16 @@ void JointReaction::setupReactionList()
             if (_inFrame.size()) {
                 expressedIn = (i < _inFrame.size()) ? _inFrame[i] : _inFrame[0];
             }
+
             if (_model->hasComponent<Frame>(expressedIn))
-                currentKey.expressedInFrame = &_model->getComponent<Frame>(expressedIn);
+                currentKey.expressedInFrame =
+                    &_model->getComponent<Frame>(expressedIn);
+            else if (_model->getBodySet().hasComponent<Frame>(expressedIn))
+                currentKey.expressedInFrame =
+                    &_model->getBodySet().getComponent<Frame>(expressedIn);
+            else if (_model->getJointSet().hasComponent<Frame>(expressedIn))
+                currentKey.expressedInFrame =
+                    &_model->getJointSet().getComponent<Frame>(expressedIn);
             else {
```

Comment:
Why was this necessary? Was it a bug fixed along the way?

Reply:
Thanks @aseth1 Would it make sense to move this sequence to one function to locate Frame then so that this logic doesn't need to be repeated elsewhere and so it can be updated if Frames can be located in other places?

Reply:
I don't think it is worth making more changes to this Analysis. There is more control using Outputs, which Joint has for reaction forces are part of. The change preserves the functionality of the `JointReaction` Analysis.